File: day-2/part-1.lua
local open = io.open

local function read_file(path)
    local file = open(path, "rb")
    if not file then return nil end
    local lines = file:lines()
    return lines
end

local file_content = read_file("day-2/input");

local valid = 0

for line in file_content do
    min, max, char, password = string.match(line, "(%d+)-(%d+) (%a+): (%a+)")
    _, count = string.gsub(password, char, "")
    if (count >= tonumber(min) and count <= tonumber(max)) then
        valid = valid + 1
    end
end

print(valid)

File: day-2/part-1.py
import os
import re

dirname = os.path.dirname(__file__)
filename = os.path.join(dirname, 'input')

input = open(filename, "r")

valid = 0

for line in input.readlines():
    match = re.search(r"(\d+)-(\d+) (.+): (.+)", line)
    min, max, char, password = match.groups()
    count = password.count(char)
    if (count >= int(min) and count <= int(max)):
        valid += 1

print(valid)
